File: Word-Formed/src/com/google/corrigan/owen/wordformed/HiScoreAdapter.java
package com.google.corrigan.owen.wordformed;


import android.content.Context;
import android.graphics.Color;
import android.util.Log;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import java.util.List;

class HiScoreAdapterView extends LinearLayout {        
        public HiScoreAdapterView(Context context, 
								HiScore hiScore ) {
            super( context );

            this.setOrientation(HORIZONTAL);        
            LinearLayout.LayoutParams nameParams = 
                new LinearLayout.LayoutParams(300, 80);
            nameParams.setMargins(20, 20, 60, 1);
            
            TextView nameControl = new TextView( context );
			nameControl.setText( hiScore.getName() );
            nameControl.setTextSize(14f);
            nameControl.setTextColor(Color.WHITE);
            addView(nameControl, nameParams);       

            LinearLayout.LayoutParams scoreParams = 
                new LinearLayout.LayoutParams(160, LayoutParams.WRAP_CONTENT);
            scoreParams.setMargins(1, 1, 1, 1);

            TextView scoreControl = new TextView(context);
            scoreControl.setText( ""+hiScore.getScore());
            scoreControl.setTextSize( 14f );
           
            scoreControl.setTextColor(Color.WHITE);
            addView( scoreControl, scoreParams);
        }
}

public class HiScoreAdapter extends BaseAdapter {

    private Context context;
    private List<HiScore> hiScoreList;

    public HiScoreAdapter(Context context, List<HiScore> hiScoreList ) { 
        this.context = context;
        this.hiScoreList = hiScoreList;
    }

    public int getCount() {                        
        return hiScoreList.size();
    }

    public Object getItem(int position) { 
        return hiScoreList.get(position);
    }

    public long getItemId(int position) {  
    	return position;
    }

    public View getView(int position, View convertView, ViewGroup parent) { 
        HiScore hiScore = hiScoreList.get(position);
        return new HiScoreAdapterView(this.context, hiScore );
    }

}

File: app/src/main/java/com/google/corrigan/owen/wordformed/HiScoreAdapter.kt
package com.google.corrigan.owen.wordformed

import android.content.Context
import android.graphics.Color
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import android.widget.TextView

class HiScoreAdapterView(context: Context, hiScore: HiScore, position: Int) : LinearLayout(context) {
    init {
        orientation = HORIZONTAL
        
        val nameParams = LayoutParams(300, 80).apply {
            setMargins(20, 20, 60, 1)
        }

        val nameControl = TextView(context).apply {
            text = "${position + 1}. ${hiScore.name}"
            textSize = 14f
            setTextColor(Color.WHITE)
        }
        addView(nameControl, nameParams)

        val scoreParams = LayoutParams(160, LayoutParams.WRAP_CONTENT).apply {
            setMargins(1, 1, 1, 1)
        }

        val scoreControl = TextView(context).apply {
            text = hiScore.score.toString()
            textSize = 14f
            setTextColor(Color.WHITE)
        }
        addView(scoreControl, scoreParams)
    }
}

class HiScoreAdapter(private val context: Context, private val hiScoreList: List<HiScore>) : BaseAdapter() {
    override fun getCount(): Int = hiScoreList.size
    override fun getItem(position: Int): Any = hiScoreList[position]
    override fun getItemId(position: Int): Long = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        return HiScoreAdapterView(context, hiScoreList[position], position)
    }
}
